src/backends/mpv.moon

```diff
@@ -0,0 +1,124 @@
+class MpvBackend extends Backend
+	new: =>
+		@name = "mpv"
+
+	appendProperty: (out, property_name, option_name) =>
+		option_name = option_name or property_name
+		prop = mp.get_property(property_name)
+		if prop and prop != ""
+			append(out, {"--#{option_name}=#{prop}"})
+
+	getPlaybackOptions: =>
+		ret = {}
+		self\appendProperty(ret, "sub-ass-override")
+		self\appendProperty(ret, "sub-ass-force-style")
+		self\appendProperty(ret, "sub-auto")
+
+		-- tracks added manually (eg. drag-and-drop) won't appear on sub-files, so we
+		-- read them from the track-list.
+		for _, track in ipairs mp.get_property_native("track-list")
+			if track["type"] == "sub" and track["external"]
+				append(ret, {"--sub-files-append=#{track['external-filename']}"})
+
+		return ret
+
+	-- Turn `MpvFilter`s into command line options.
+	solveFilters: (filters) =>
+		solved = {}
+		for filter in *filters
+			str = filter.lavfiCompat and "lavfi-" or ""
+			str ..= filter.name .. "="
+			for k, v in pairs filter.params
+				if tonumber(k) == nil
+					str ..= "#{k}=%#{string.len(v)}%#{v}:"
+				else
+					str ..= "#{v}:"
```

Comment:
I don't really get this. I assume this is because some filters we use internally don't have key/value explicited (such as the colormatrix one)? I'd prefer to add the corresponding keys to the filters than to use this.

Reply:
Yeah... Admittedly, I'm an amateur when it comes to this stuff. I didn't read up on the exact specifications on how mpv interprets these multi-parameter options and it was hard to get it to behave consistently. There's that combined with how flimsily lua does tables (at least to someone unfamiliar with it like me).

Basically, that if is for single-value filter parameters like here: https://github.com/sides/mpv-webm/blob/master/src/formats/rawvideo.moon#L31-L33
mpv understands `--vf=format=yuv444p16` but it does not understand `--vf=format=@0=%9%yuv444p16`. However, examining that further, `--vf=format=yuv444p16` is just a shortcut for `--vf=format=fmt=%9%yuv444p16`, meaning the correct way to do those filters would be `MpvFilter("format", { fmt: "yuv444p16" })`, making this a catch-all format.

So, this check can be removed, but it means filters done in lua code cannot make use of certain shortcuts you can on the command-line. They also must use the `@n` syntax for keys. This supposedly guarantees order mpv-side even if lua-side `@1` comes before `@0`, for example. I.e., filters must be totally explicit (in the same format as `get_property_native("vf")` does them), which is fine though.

Now, as for ffmpeg...